spec/range-spec.coffee

```diff
@@ -2,6 +2,139 @@ Range = require "../src/range"
 Point = require "../src/point"
 
 describe "Range", ->
+  describe "::copy()", ->
+    it "returns a copy of the given range", ->
+      expect(Range([1, 3], [3, 4]).copy().isEqual(Range([1, 3], [3, 4])))
+      expect(Range([1, 3], [2, 3]).copy().isEqual([[1, 2], [2, 3]]))
+
+  describe "::negate()", ->
+    it "should negate the start and end points", ->
+      expect(Range([ 0,  0], [ 0,  0]).negate().isEqual([[ 0,  0], [ 0,  0]]))
+      expect(Range([ 1,  2], [ 3,  4]).negate().isEqual([[-1, -2], [-3, -4]]))
+      expect(Range([-1, -2], [-3, -4]).negate().isEqual([[ 1,  2], [ 3,  4]]))
+      expect(Range([-1,  2], [ 3, -4]).negate().isEqual([[ 1, -2], [-3,  4]]))
+
+  describe "::reverse()", ->
+    it "returns a range with reversed start and end points", ->
+      expect(Range(Point(0, 2), Point(3, 4)).reverse().isEqual([[3, 4], [0, 2]]))
+      expect(Range([3, 4], [2, 3]).reverse().isEqual([[2, 3], [3, 4]]))
+
+  describe "::isEmpty()", ->
+    it "returns whether if start is equal to end", ->
+      expect(Range([0, 0], [0, 0]).isEmpty()).toBe true
+      expect(Range([2, 3], [4, 5]).isEmpty()).toBe false
+
+  describe "::isSingleLine()", ->
+    it "returns whether start row is equal to end row", ->
+      expect(Range([2, 3], [3, 4]).isSingleLine()).toBe false
+      expect(Range([1, 2], [1, 5]).isSingleLine()).toBe true
+
+  describe "::getRowCount()", ->
+    it "returns total number of rows in the given range", ->
+      expect(Range([2, 3], [4, 4]).getRowCount() is 3)
+      expect(Range([2, 4], [2, 6]).getRowCount() is 1)
+      expect(Range([2, 4], [2, 1]).getRowCount() is 1)
+
+      expect(Range([2, 5], [0, 5]).getRowCount() is -2)
```

Comment:
I'm not sure at this point. When in doubt though, we need to preserve the `text-buffer` API so that we can substitute this module for that one without changing the behavior of any Atom packages.